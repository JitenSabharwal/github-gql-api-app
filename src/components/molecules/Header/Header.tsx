import React from "react"
import styled from "styled-components"
import Icon from "../../atoms/Icon/Icon"
import Logo from "../../../logo.svg"

const Root = styled.header`
  display: flex;
  background: #24292e;
  color: #ffffffe6;
  padding: 8px 24px;
`
const Heading = styled.div`
  text-align: center;
  font-size: 24px;
  font-weight: bold;
  line-height: 36px;
  width: 100%;
`
const BackButton = styled.div`
  display: flex;
  justify-content: flex-start;
  margin-left: 16px;
  line-height: 24px;
  align-items: center;
  cursor: pointer;
`
const ButtonText = styled.span`
  font-size: 13px;
  width: max-content;
`

export default function Header({
  onBackClick,
  heading,
}: {
  onBackClick?: () => void
  heading: string
}) {
  return (
    <Root>
      {onBackClick && (
        <>
          <BackButton onClick={onBackClick}>
            {/* <Icon icon={Logo} /> */}
            <ButtonText>{"Back to Search"}</ButtonText>
          </BackButton>
        </>
      )}
      <Heading>{heading}</Heading>
    </Root>
  )
}
