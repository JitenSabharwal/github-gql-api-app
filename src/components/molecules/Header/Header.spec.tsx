import Header from "./index"
import React from "react"
import { shallow } from "enzyme"

describe("Header", () => {
  let container: React.ReactElement
  beforeAll(() => (container = <Header heading={"Heading"} />))
  it("Should render Header", () => {
    const component = shallow(container)
  })
  it("Should render initial layout", () => {
    const component = shallow(container)
    expect(component.getElements()).toMatchSnapshot()
  })

  it("Should render initial layout with back button", () => {
    const component = shallow(
      <Header heading={"Heading"} onBackClick={() => {}} />
    )
    expect(component.getElements()).toMatchSnapshot()
  })
})
