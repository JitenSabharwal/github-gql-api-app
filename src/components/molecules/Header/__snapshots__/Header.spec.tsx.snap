// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should render initial layout 1`] = `
Array [
  <styled.header>
    <styled.div>
      Heading
    </styled.div>
  </styled.header>,
]
`;

exports[`Header Should render initial layout with back button 1`] = `
Array [
  <styled.header>
    <React.Fragment>
      <styled.div
        onClick={[Function]}
      >
        <styled.span>
          Back to Search
        </styled.span>
      </styled.div>
    </React.Fragment>
    <styled.div>
      Heading
    </styled.div>
  </styled.header>,
]
`;
